### PR TITLE
Add frontend metrics to default metrics configuration

### LINKED ISSUE #2133: TorchServe default metrics config yaml file
### 🐛 Describe the bug

Update existing metrics yaml config file to include TorchServe frontend existing metrics.

### Error logs

N/A

### Installation instructions

N/A

### Model Packaing

N/A

### config.properties

_No response_

### Versions

N/A

### Repro instructions

N/A

### Possible Solution

_No response_

### PR DESCRIPTION
**Pull request by namannandan**
## Description

Add frontend metrics collected by [system_metrics.py](https://github.com/pytorch/serve/blob/master/ts/metrics/system_metrics.py) to [default metrics configuration file](https://github.com/pytorch/serve/blob/master/ts/configs/metrics.yaml)

Fixes #2133

### CODE REVIEW

**Review by lxning — COMMENTED** [review]
Please search "new Metric" in frontend java code, add all the metrics generated from the metrics' objects.

File: ts/configs/metrics.yaml [hunk 1]
```diff
@@ -11,13 +12,46 @@ ts_metrics:
     - name: GaugeTsMetricExample
       unit: ms
       dimensions: [*model_name, *level]
```
Comment (lxning) [inline ts/configs/metrics.yaml:14]: can you point out the code using this metrics?

File: ts/configs/metrics.yaml [hunk 2]
```diff
@@ -11,13 +12,46 @@ ts_metrics:
...
+    - name: GPUMemoryUsed
+      unit: MB
+      dimensions: [*level, *device_id]
+    - name: GPUUtilization
+      unit: percent
+      dimensions: [*level, *device_id]
   histogram:
     - name: HistogramTsMetricExample
       unit: ms
       dimensions: [*model_name, *level]
```
Comment (lxning) [inline ts/configs/metrics.yaml:48]: can you point out the code using this metrics?

File: ts/configs/metrics.yaml [hunk 3]
```diff
@@ -11,13 +12,46 @@ ts_metrics:
...
 
 model_metrics:
   counter:
+    - name: CounterModelMetricNameExample
+      unit: ms
+      dimensions: [*model_name, *level]
```
Comment (lxning) [inline ts/configs/metrics.yaml:54]: can you point out the code using this metrics?

**Comment by lxning** [comment]
@namannandan Please 
- first clean up unused metric in the existing metrics.yaml file.
- second search all existing metrics in frontend and backend code, add them into ts_metrics or model_metrics in the metrics.yaml.

**Comment by namannandan** [comment]
Addressed review comments.
Skipped [WorkerLoadTime](https://github.com/pytorch/serve/blob/86d440041b663961c71a6262fe648111d85b27d8/frontend/server/src/main/java/org/pytorch/serve/wlm/WorkerThread.java#L163) metric since the metric name is not fixed and is [determined at runtime](https://github.com/pytorch/serve/blob/86d440041b663961c71a6262fe648111d85b27d8/frontend/server/src/main/java/org/pytorch/serve/wlm/WorkerThread.java#L164).

File: ts/configs/metrics.yaml [hunk 4]
```diff
@@ -1,34 +1,62 @@
...
+      unit: percent
+      dimensions: [*level, *device_id]
+    - name: GPUMemoryUsed
+      unit: MB
+      dimensions: [*level, *device_id]
+    - name: GPUUtilization
+      unit: percent
+      dimensions: [*level, *device_id]
 
 model_metrics:
```
Comment (lxning) [inline ts/configs/metrics.yaml:55]: Add comment in the file.
dimension "hostname" is automatically added for model metrics at backend.

File: ts/configs/metrics.yaml [hunk 5]
```diff
@@ -1,34 +1,62 @@
 dimensions:
-  - &model_name "model_name"
+  - &model_name "ModelName"
   - &level "Level"
+  - &device_id "device_id"
 
 ts_metrics:
```
Comment (lxning) [inline ts/configs/metrics.yaml:6]: missing hostname in all ts_metrics

**Review by lxning — COMMENTED** [review]
the following parts are missing:

- mode (check design doc)
- metrics defined in [prometheus](https://github.com/pytorch/serve/blob/master/frontend/server/src/main/java/org/pytorch/serve/metrics/format/prometheous/PrometheusMetricManager.java)
- WorkerLoadTime metric

File: ts/configs/metrics.yaml (hunk 5 above)
Comment (namannandan) [reply ts/configs/metrics.yaml:12]: Added hostname dimensioin

**Comment by namannandan** [comment]
Addressed review comments:
- added `mode` configuration
- added metrics defined in [prometheus](https://github.com/pytorch/serve/blob/master/frontend/server/src/main/java/org/pytorch/serve/metrics/format/prometheous/PrometheusMetricManager.java)
- added `WorkerLoadTime` metric. Since the metric name is determined at [runtime](https://github.com/pytorch/serve/blob/86d440041b663961c71a6262fe648111d85b27d8/frontend/server/src/main/java/org/pytorch/serve/wlm/WorkerThread.java#L164), included a dimension called `WorkerName` for this metric.

File: ts/configs/metrics.yaml [hunk 6]
```diff
@@ -1,34 +1,80 @@
...
+      dimensions: [*level, *hostname]
+    - name: InferenceRequestsTotal
+      unit: Count
+      dimensions: [*model_name, *model_version, *hostname]
+    - name: CumulativeInferenceLatencyMicroseconds
+      unit: Microseconds
+      dimensions: [*model_name, *model_version, *hostname]
+    - name: CumulativeQueueLatencyMicroseconds
+      unit: Microseconds
+      dimensions: [*model_name, *model_version, *hostname]
```
Comment (lxning) [inline ts/configs/metrics.yaml:30]: Please use original prometheus metric name, not help.

**Review by lxning — APPROVED** [review]

**Review by msaroufim — APPROVED** [review]

**Pull request merged**